x-pack/test/observability_ai_assistant_api_integration/tests/knowledge_base/knowledge_base_user_instructions.spec.ts

```diff
@@ -362,5 +365,64 @@ export default function ApiTest({ getService }: FtrProviderContext) {
         expect(res2).to.be('');
       });
     });
+
+    describe('security roles and access privileges', () => {
+      const routes: Array<{
+        endpoint: string;
+        method: 'get' | 'post' | 'put' | 'delete';
+        payload?: Record<string, any>;
+        requiredPrivileges: string[];
+      }> = [
+        {
+          endpoint: '/internal/observability_ai_assistant/kb/user_instructions',
+          method: 'put',
+          payload: {
+            body: {
+              id: 'test-instruction',
+              text: 'Test user instruction',
+              public: true,
+            },
+          },
+          requiredPrivileges: ['ai_assistant'],
+        },
+        {
+          endpoint: '/internal/observability_ai_assistant/kb/user_instructions',
+          method: 'get',
+          requiredPrivileges: ['ai_assistant'],
+        },
```

Comment:
I think it would be better to avoid the dynamic behaviour and simply have separate specs per endpoint. WDYT about this suggestion? Note: Completely untested and I don't remember whether it's `e.status` or `e.statusCode` or something else. 

```suggestion
it('GET /internal/observability_ai_assistant/kb/user_instructions', () => {
  expect.assertions(1)
  try {
    await observabilityAIAssistantAPIClient.noAuth({
      endpoint: "GET /internal/observability_ai_assistant/kb/user_instructions"
    });
  } catch(e) {
    expect(e.status).toBe(403)
  }
})
```

Reply:
Updated